tests/chainer_tests/functions_tests/array_tests/test_rollaxis.py

```diff
@@ -0,0 +1,55 @@
+import unittest
+
+import numpy
+
+import chainer
+from chainer import cuda
+from chainer import functions
+from chainer import gradient_check
+from chainer import testing
+from chainer.testing import attr
+
+
+@testing.parameterize(
+    {'axis': 0, 'start': 2, 'out_shape': (3, 2, 4)},
+    {'axis': 2, 'start': 0, 'out_shape': (4, 2, 3)},
+    {'axis': 1, 'start': 1, 'out_shape': (2, 3, 4)},
+    {'axis': -3, 'start': 2, 'out_shape': (3, 2, 4)},
+    {'axis': -1, 'start': 0, 'out_shape': (4, 2, 3)},
+    {'axis': -2, 'start': -2, 'out_shape': (2, 3, 4)},
```

Comment:
I don't think we need to test all patterns. We need to remove non-trivial logic from the test code as possible as we can.